gateway/src/service.rs

```diff
@@ -1023,6 +1040,62 @@ impl DockerContext for GatewayContext {
     fn container_settings(&self) -> &ContainerSettings {
         &self.settings
     }
+
+    #[instrument(name = "getting container stats from the proper source", skip_all)]
```

Comment:
I think we should record these fields.
```suggestion
    #[instrument(name = "getting container stats from the proper source", skip_all, fields(docker_stats_source = %self.docker_stats_source, shuttle.container.id = container_id))]
```